lib/core/locks.nim

```diff
@@ -19,12 +19,16 @@ when not compileOption("threads") and not defined(nimdoc):
     {.error: "Locks requires --threads:on option.".}
 
 const insideRLocksModule = false
+const useOrcArc = defined(gcArc) or defined(gcOrc)
+
 include "system/syslocks"
 
 type
-  Lock* = SysLock ## Nim lock; whether this is re-entrant
+  Lock* = object
+    lock: SysLock ## Nim lock; whether this is re-entrant
                   ## or not is unspecified!
```

Comment:
```suggestion
  Lock* = object
    ## Nim lock; whether this is re-entrant or not is unspecified!
    lock: SysLock
```
Otherwise the doc comment is attached to the field.

Reply:
thanks